src/core/geometry_utils.h

```diff
@@ -11,8 +11,35 @@
 #include <cmath>
 
 
-#define EQ_DOUBLE(a, b)                         \
-  (std::abs((a) - (b)) < 0.0001)
+inline bool equal(double a, double b) {
+  return (std::abs((a) - (b)) < 0.0001);
+}
+
+inline bool equal(double a, double b, double c) {
+  return (equal(a, b) && equal(a, c));
+}
+
+inline bool less_or_equal(double a, double b) {
+  return (equal(a, b) || (a < b));
+}
+
+inline bool is_between(double point, double left, double right) {
+  return (less_or_equal(left, point) && less_or_equal(point, right)) ||
```

Comment:
1. If we need to check both cases ((l < r) and (r < l)) the arg names seem confusing to me (the "left" is supposed to be less than the "right").
2. If you choose to rename args please align the OR clauses.